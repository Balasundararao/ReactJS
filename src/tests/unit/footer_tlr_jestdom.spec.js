import React from 'react';
import ReactDOM from 'react-dom';
import FooterComponent from '../../components/footer/footercomponent';

// testing library react....
import {render, cleanup} from '@testing-library/react';

//snapshot testing....
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Smoke Test , Renders with out crashing", () => {
    const div = document.createElement('div');
    ReactDOM.render('<Button>', div);
})

it('Matches SnapShot ', () => {
    const tree = renderer.create(<FooterComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
})
