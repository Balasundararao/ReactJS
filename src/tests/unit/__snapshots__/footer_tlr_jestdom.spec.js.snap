// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches SnapShot  1`] = `
<footer
  className="footer"
>
  <div>
    Copyright Â© 
    2020
     Apple Inc. All rights reserved..
  </div>
  <nav
    className="navbar_bottom"
  >
    <ul>
      <li
        className="delimeter"
      >
        <a
          alt="crumb"
          className="abc"
          href="/compute/support/"
        >
          Support
        </a>
      </li>
      <li>
        <a
          href="rdar://new/problem/component=ireporter&version=Whistler"
        >
          File a Radar 
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;
